serpentTools/parsers/detector.py

```diff
@@ -71,7 +73,17 @@ def _addDetector(self, chunk, detName, binType):
                        .format(binType, detName))
 
     def _precheck(self):
-        pass
+        """ Count how many detectors are in the file
+        """
+        with open(self.filePath) as fh:
+            for line in fh:
+                sline = line.split()
+                if sline == []:
+                    continue
+                elif 'DET' in sline[0]:
+                    self.detCount += 1
 
     def _postcheck(self):
-        pass
+        if self.detCount != len(self.detectors):
+            error("Did not parse expected number of detectors in file\n"
+                  "{}".format(self.filePath))

```

Comment:
Similarly, I think it is sufficient to just check `if not self.detectors` because the user could request to only store a subset of detectors with the [`detector.names`](https://github.com/CORE-GATECH-GROUP/serpent-tools/blob/master/serpentTools/settings.py#L55) setting

Reply:
Alright, for the detectors, I just removed the postcheck for now.